src/libsystemd-network/ndisc-router.c

```diff
@@ -468,6 +474,13 @@ static int get_route_info(sd_ndisc_router *rt, uint8_t **ret) {
         if (ri[2] > 128)
                 return -EBADMSG;
 
+
+        /* RFC4191 – Section 2.3 Route Information Option's Prf field:
+         * If the Reserved (10) value is received, the Route Information Option MUST be ignored.
+         */
+        if (!IN_SET((ri[3] >> 3) & 3, SD_NDISC_PREFERENCE_LOW, SD_NDISC_PREFERENCE_HIGH))
+                return -EBADMSG;
```

Comment:
This causes reading router information failed, and networkd make the link enter failed state.

So, instead, please make `sd_ndisc_router_route_get_preference()` return a recognizable errno e.g. `-EOPNOTSUPP`, then make networkd handle it gracefully.

Or, keep `sd_ndisc_router_route_get_preference()` as is, and make networkd silently ignore the option when the value is `10b`. This may be simpler.